extensions/cactus-plugin-htlc-coordinator/src/test/typescript/integration/plugin-htlc-coordinator/counterparty-htlc-endpoint.test.ts

```diff
@@ -0,0 +1,264 @@
+    // signing credential pointing to this keychain entry.
+    backend: new Map([[TestTokenJSON.contractName, TestTokenJSON]]),
+    logLevel,
+  });
+  keychainPlugin.set(DemoHelperJSON.contractName, DemoHelperJSON);
+  keychainPlugin.set(HashTimeLockJSON.contractName, HashTimeLockJSON);
+
+  const factory = new PluginFactoryLedgerConnector({
+    pluginImportType: PluginImportType.Local,
+  });
+
+  const pluginRegistry = new PluginRegistry({});
+  const connector: PluginLedgerConnectorBesu = await factory.create({
+    rpcApiHttpHost,
+    rpcApiWsHost,
+    logLevel,
+    instanceId: connectorId,
+    pluginRegistry: new PluginRegistry({ plugins: [keychainPlugin] }),
+  });
+
+  const httpServer = createServer();
+  await new Promise((resolve, reject) => {
+    httpServer.once("error", reject);
+    httpServer.once("listening", resolve);
+    httpServer.listen(0, "127.0.0.1");
+  });
+
+  pluginRegistry.add(connector);
+  const pluginOptions: IPluginHTLCCoordinatorOptions = {
+    instanceId: uuidv4(),
+    logLevel,
+    pluginRegistry,
+  };
+
+  const factoryHTLC = new PluginFactoryHTLCCoordinator({
+    pluginImportType: PluginImportType.Local,
+  });
+  const pluginHTLCCoordinator = await factoryHTLC.create(pluginOptions);
+  pluginRegistry.add(pluginHTLCCoordinator);
+
+  const expressApp = express();
+  expressApp.use(bodyParser.json({ limit: "250mb" }));
+  const server = http.createServer(expressApp);
+  const listenOptions: IListenOptions = {
+    hostname: "0.0.0.0",
+    port: 0,
+    server,
+  };
+  const addressInfo = (await Servers.listen(listenOptions)) as AddressInfo;
+  test.onFinish(async () => await Servers.shutdown(server));
+  const { address, port } = addressInfo;
+  const apiHost = `http://${address}:${port}`;
+
+  const configuration = new Configuration({ basePath: apiHost });
+  const api = new BesuApi(configuration);
+  await pluginHTLCCoordinator.getOrCreateWebServices();
+  await pluginHTLCCoordinator.registerWebServices(expressApp);
+  t.comment("Deploys TestToken via .json file on deployContract function");
+  const deployOutToken = await connector.deployContract({
+    contractName: TestTokenJSON.contractName,
+    contractAbi: TestTokenJSON.abi,
+    bytecode: TestTokenJSON.bytecode,
+    web3SigningCredential,
+    keychainId,
+    constructorArgs: ["100", "token", "2", "TKN"],
+    gas: estimatedGas,
+  });
+  t.ok(deployOutToken, "deployContract() output is truthy OK");
+  t.ok(
+    deployOutToken.transactionReceipt,
+    "deployContract() output.transactionReceipt is truthy OK",
+  );
+  t.ok(
+    deployOutToken.transactionReceipt.contractAddress,
+    "deployContract() output.transactionReceipt.contractAddress is truthy OK",
+  );
+  const tokenAddress = deployOutToken.transactionReceipt
+    .contractAddress as string;
+
+  t.comment("Approve 10 Tokens to HashTimeLockAddress");
+  const { success } = await connector.invokeContract({
```

Comment:
@jagpreetsinghsasan Please assert more specifically the status code being 2xx and also refactor this to use the API client not the connector instance directly.

Reply:
Here, the apiClient doesnt have methods to invoke the contract (I think so)
![image](https://user-images.githubusercontent.com/56861721/123908830-be3dae80-d995-11eb-9210-c5f7f3f1da4f.png)

So should I be changing the openapi specs to include the invokeContract and then define the endpoint and use it?

Reply:
@jagpreetsinghsasan No need to change the openapi specs you just need an instance of the connector's API client (the one you are looking at on that screenshot looks like it's the coordinator plugin's)

Off-topic: Please tag my username when sending a comment otherwise the GitHub notification mechanism doesn't work as well (I had to scroll through all the messages here to find the one where you were commenting - if you actually had multiple comments then also let me know I only saw this one on my notifications inbox)

Reply:
Sure @petermetz , I will have a look on the same.